src/appearance.cpp

```diff
@@ -192,4 +194,5 @@ void Appearance::setDockTransparency(bool enabled)
 
     m_dockTransparency = enabled;
     m_dockSettings->setValue("DockTransparency", enabled);
+    system("killall cyber-dock; cyber-dock &");
```

Comment:
for spawning the dock, could you use QProcess here instead? refer to the code of cyber-session for an example

ideally you could also use a Qt wrapper for killing a process but im not sure if there are any

Reply:
Done.